x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/assignment/TrainedModelAssignmentClusterService.java

```diff
@@ -242,14 +245,54 @@ static ClusterState removeRoutingToUnassignableNodes(ClusterState currentState)
                         routedNodeIdsToRemove
                     )
                 );
-                TrainedModelAssignment.Builder assignmentBuilder = TrainedModelAssignment.Builder.fromAssignment(assignment);
-                routedNodeIdsToRemove.forEach(assignmentBuilder::removeRoutingEntry);
+
+                TrainedModelAssignment.Builder assignmentBuilder = removeRoutingBuilder(
```

Comment:
`TrainedModelAssignmentClusterService` runs on the master node so during a rolling upgrade from 8.3.x once the master node has been upgraded to 8.11.0 this code will be hit if there are still nodes in the cluster with version < 8.4.0. 

The old code just removed the route, the new code sets it to stopping, either way if you call _infer the request won't be routed to that node. From that perspective it makes no difference but I think I missed a detail. If the shutting down node is >= 8.11.0 then that node will stop gracefully instead of stoping abruptly.

In the comment I was suggesting that the code will not have any effect so better to remove it but now I think there is a case where it will make a difference and that is in this scenario:
- Rolling upgrade from < 8.4.0
- Mixed cluster with 8.11.0 and 8.4.0 nodes 
- Master node has been upgraded to 8.11.0
- An ML node upgraded to 8.11.0 is about to be shutdown as notified by the shutdown cluster state change
- That 8.11.0 ML node can stop gracefully rather than abruptly